packages/wonder-blocks-announcer/src/Announcer.ts

```diff
@@ -0,0 +1,257 @@
+
+const REMOVAL_TIMEOUT_DELAY = 5000;
+const DEFAULT_WAIT_THRESHOLD = 250;
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
+    private static _instance: Announcer | null;
+    node: HTMLElement | null = null;
+    regionFactory: RegionFactory = {
+        count: 2,
+        aIndex: 0,
+        pIndex: 0,
+    };
+    dictionary: RegionDictionary = new Map();
+    stateLock = false;
+
+    private constructor() {
+        if (typeof document !== "undefined") {
+            const topLevelId = `wbAnnounce`;
+            // Check if our top level element already exists
+            const announcerCheck = document.getElementById(topLevelId);
+
+            // Init new structure if the coast is clear
+            if (announcerCheck === null) {
+                this.init(topLevelId);
+            }
+            // The structure exists but references are lost, so help HMR recover
+            else {
+                this.reattachNodes();
+            }
+        }
+    }
+    /**
+     * Singleton handler to ensure we only have one Announcer instance
+     * @returns {Announcer}
+     */
+    static getInstance() {
+        if (!Announcer._instance) {
+            Announcer._instance = new Announcer();
+        }
+        return Announcer._instance;
+    }
+    /**
+     * Internal initializer method to create live region elements
+     * Prepends regions to document body
+     * @param {string} id ID of the top level node (wbAnnounce)
+     */
+    init(id: string) {
+        this.node = document.createElement("div");
+        this.node.id = id;
+        this.node.setAttribute("data-testid", id);
+
+        Object.assign(this.node.style, srOnly);
+
+        // For each level, we create at least two live region elements.
+        // This is to work around AT occasionally dropping messages.
+        const aWrapper = createRegionWrapper("assertive");
+        createDuplicateRegions(
+            aWrapper,
+            "assertive",
+            this.regionFactory.count,
+            this.dictionary,
+        );
+        this.node?.appendChild(aWrapper);
+
+        const pWrapper = createRegionWrapper("polite");
+        createDuplicateRegions(
+            pWrapper,
+            "polite",
+            this.regionFactory.count,
+            this.dictionary,
+        );
+        this.node.appendChild(pWrapper);
+
+        document.body.append(this.node);
+    }
+    /**
+     * Recover in the event regions get lost
+     * This happens in Storybook or other HMR environments when saving a file:
+     * Announcer exists, but it loses the connection to DOM element Refs
+     */
+    reattachNodes() {
+        const announcerCheck = document.getElementById(`wbAnnounce`);
+        if (announcerCheck !== null) {
+            this.node = announcerCheck;
+            const regions = Array.from(
+                announcerCheck.querySelectorAll<HTMLElement>(
+                    "[id^='wbARegion'",
```

Comment:
I think this line is missing a `]`

Reply:
Good call! I wish there was some more static typing for those strings...